rescue/src/util.rs

```diff
@@ -0,0 +1,45 @@
+use gcd::Gcd;
+use modinverse::modinverse;
+use num::{BigUint, One};
+use p3_field::PrimeField64;
+use sha3::{
+    digest::{ExtendableOutput, Update, XofReader},
+    Shake256,
+};
+
+/// The binomial function from combinatorics, used to compute the number of rounds needed for soundness.
+pub(crate) fn binomial(n: usize, k: usize) -> BigUint {
```

Comment:
Looks like there's `binomial` in `num_integer`, could we use that?

Reply:
Ohh right, I initially didn't do that because `Integer` isn't implemented for `U256`, but it works for BigUInt